samples/authentication/serial/auth_client/src/main.c

```diff
@@ -0,0 +1,239 @@
+				.priv_key_size = sizeof(auth_client_privatekey)
+			}
+		}
+	}
+};
+#endif
+
+#if defined(CONFIG_AUTH_CHALLENGE_RESPONSE)
+#define NEW_SHARED_KEY_LEN          (32u)
+
+/* Use a different key than default */
+static uint8_t chal_resp_sharedkey[NEW_SHARED_KEY_LEN] = {
+	0x21, 0x8e, 0x37, 0x42, 0x1e, 0xe1, 0x2a, 0x22, 0x7c, 0x4b, 0x3f, 0x3f, 0x07, 0x5e, 0x8a, 0xd8,
+	0x24, 0xdf, 0xca, 0xf4, 0x04, 0xd0, 0x3e, 0x22, 0x61, 0x9f, 0x24, 0xa3, 0xc7, 0xf6, 0x5d, 0x66
+};
+
+
+static struct auth_optional_param chal_resp_param = {
+	.param_id = AUTH_CHALRESP_PARAM,
+	.param_body = {
+		.chal_resp = {
+			.shared_key = chal_resp_sharedkey,
+		},
+	}
+};
+#endif
+
+
+/* Authentication connection info */
+static struct authenticate_conn auth_conn_serial;
+
+/**
+ * Authentication status callback.
+ *
+ * @param auth_conn   The authentication connection.
+ * @param instance    Authentication instance id.
+ * @param status      Authentication status.
+ * @param context     Optional context
+ */
+void auth_status_callback(struct authenticate_conn *auth_conn,
+			  enum auth_instance_id instance,
+			  enum auth_status status, void *context)
+{
+	printk("Authentication instance (%d) status: %s\n", instance,
+	       auth_lib_getstatus_str(status));
+
+#if defined(CONFIG_AUTH_DTLS)
+	if (status == AUTH_STATUS_IN_PROCESS) {
+		printk("     DTLS may take 30-60 seconds.\n");
+	}
+#endif
+
+
+	if ((status == AUTH_STATUS_FAILED) ||
+	    (status == AUTH_STATUS_AUTHENTICATION_FAILED) ||
+	    (status == AUTH_STATUS_SUCCESSFUL)) {
+		/* Authentication has finished */
+		auth_lib_deinit(auth_conn);
+	}
+}
+
+/**
+ * Process log messages
+ */
+static void process_log_msgs(void)
+{
+	while (log_process(false)) {
+		; /* intentionally empty statement */
+	}
+}
+
+/**
+ * Idle processing.
+ */
+static void idle_process(void)
+{
+	/* Just spin while the BT modules handle the connection. */
+	while (true) {
+
+		process_log_msgs();
+
+		/* Let the handshake thread run */
+		k_yield();
+	}
+}
+
+
+/**
+ * Configures the UART
+ *
+ * @return 0 on success, else negative error code.
+ */
+static int config_uart(void)
+{
+	struct auth_xp_serial_params xp_params;
+
+	uart_dev = device_get_binding(DT_LABEL(DT_NODELABEL(uart0)));
+
+	int err = uart_configure(uart_dev, &uart_cfg);
+
+	if (err) {
+		LOG_ERR("Failed to configure UART, err: %d", err);
+		return err;
+	}
+
+	/* If successful,then init lower transport layer. */
+	xp_params.uart_dev = uart_dev;
+
+	err = auth_xport_init(&auth_conn_serial.xport_hdl,
+			      auth_conn_serial.instance,
+			      AUTH_XP_TYPE_SERIAL, &xp_params);
+
+	if (err) {
+		LOG_ERR("Failed to initialize authentication transport, error: %d", err);
+	}
+
+	return err;
+}
+
+
+void main(void)
+{
+	struct auth_optional_param *opt_parms = NULL;
+
+	log_init();
+
+#if defined(CONFIG_AUTH_DTLS) && defined(CONFIG_AUTH_CHALLENGE_RESPONSE)
+#error Invalid authentication config, either DTLS or Challenge-Response, not both.
+#endif
```

Comment:
This kind of logic should be in the build system. These options have to be mutually exclusive

Reply:
This is actually an error, removed code.